ledger/streaming/api/src/components.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use mc_attest_core::VerificationReport;
+use mc_consensus_scp::QuorumSet;
+use mc_transaction_core::BlockData;
+
+/// Wrapper for the components needed for block streaming.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct BlockStreamComponents {
+    /// The block data.
+    pub block_data: BlockData,
+    /// The SCP quorum set for this block.
+    /// Currently optional, will become required with a future BlockVersion.
+    pub quorum_set: Option<QuorumSet>,
+    /// The AVR for this block.
+    /// Currently optional, will become required with a future BlockVersion.
+    pub verification_report: Option<VerificationReport>,
+}
```

Comment:
Seems like this could possibly be merged with components.rs?

Reply:
I don't understand your comment. This is `components.rs`...

Reply:
I suspect consumers are eventually going to want a NodeID in the component, but I understand the motivation behind not having it. I'm not entirely sure that the overhead in memory/speed would be enough of a con against including it in the component object for the convenience of tracking, error handling, and business logic. Having to run an SCP discovery predicate/patch it into a stream every time you want to know the node you're dealing with sounds bad from a maintainability perspective.

It would be interesting to clock the pipeline speed with NodeID to see if makes any appreciable difference.

Other than that, this is good.